js/store.js

```diff
@@ -366,6 +409,10 @@ const store = {
 
   async addFirstPartyLink(firstPartyHostname, thirdPartyHostname) {
     const firstParty = await this.getWebsite(firstPartyHostname);
+    // We are likely storing the tp first, lets account for that
```

Comment:
Ah, because the `setFirstParty` doesn't write the first party in the database until after the tab has loaded (for visible tabs), we may get an empty object for a while from storage.

Since we call `setFirstParty` from `setThirdParty`, we could store this information sooner than waiting for `setFirstParty` to execute?

Reply:
Yep, update was getting called with no `firstParty` key which made the graph broken on update view.